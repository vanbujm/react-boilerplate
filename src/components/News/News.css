
.newsItem {
  margin: 0 0 2rem;
}

.newsTitle {
  font-size: 1.5rem;
}

.newsDesc {
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-size: 1.125rem;
  }

  pre {
    white-space: pre-wrap;
    font-size: 0.875rem;
  }

  img {
    max-width: 100%;
  }
}
