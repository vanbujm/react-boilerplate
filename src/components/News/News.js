import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import s from './News.css';

export class NewsComponent extends React.Component {
  static propTypes = {
    news: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        link: PropTypes.string.isRequired,
        content: PropTypes.string,
      }),
    ).isRequired,
  };

  render() {
    const renderedNews = this.props.news.map(item => (
      <article key={item.link} className={s.newsItem}>
        <h1 className={s.newsTitle}>
          <a href={item.link}>{item.title}</a>
        </h1>
        <div
          className={s.newsDesc}
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: item.content }}
        />
      </article>
    ));
    return <div>{renderedNews}</div>;
  }
}

export default withStyles(s)(NewsComponent);
