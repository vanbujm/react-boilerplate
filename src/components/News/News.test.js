/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import { NewsComponent } from './News';

const fakeNews = [
  {
    content: '<h1>Some mock content with html </h1>',
    link: 'https://reactjsnews.com/fake-news',
    title: 'Using Fake News',
  },
];

describe('News', () => {
  test('renders correctly', () => {
    const wrapper = renderer.create(<NewsComponent news={fakeNews} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
