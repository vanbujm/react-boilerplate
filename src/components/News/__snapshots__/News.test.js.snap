// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News renders correctly 1`] = `
<article
  className="newsItem"
>
  <h1
    className="newsTitle"
  >
    <a
      href="https://reactjsnews.com/fake-news"
    >
      Using Fake News
    </a>
  </h1>
  <div
    className="newsDesc"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>Some mock content with html </h1>",
      }
    }
  />
</article>
`;
